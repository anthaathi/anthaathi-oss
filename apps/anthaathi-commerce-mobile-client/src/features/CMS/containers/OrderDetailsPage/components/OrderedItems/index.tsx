import {Image, View, Text as RNText, Pressable} from 'react-native';
import React from 'react';
import {Divider, Text, useTheme} from 'react-native-paper';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {useIntl} from 'react-intl';
import {OrderDetailsPageComponentType} from '../../../../types/common';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {ProductProps} from '../../../ProductListPage/components/ProductList';

export interface ItemProps extends ProductProps {
  numberOfItems: number;
}

export interface OrderedItemProps {
  title: string;
  items: ItemProps[];
  handlePress?: (item: ProductProps) => void;
}

const OrderedItems = (props: OrderedItemProps) => {
  const theme = useTheme();
  const itemHeight = useResponsiveValue([120, 250, 290, 330]);
  const itemWidth = useResponsiveValue([120, 240, 280, 320]);

  return (
    <View
      style={{
        marginHorizontal: 10,
      }}
      testID="orderedItems">
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}>
        <RNText
          style={{
            color: theme.colors.titleTextColor,
            fontSize: 16,
            fontWeight: '600',
            marginBottom: 5,
          }}>
          {props.title}
        </RNText>
      </View>

      <View
        style={{
          backgroundColor: '#fff',
          paddingVertical: 5,
          paddingHorizontal: 5,
          borderColor: theme.colors.cardBorderColor,
          borderWidth: 1,
          borderRadius: 4,
        }}>
        {props.items.map((item, index) => {
          return (
            <ItemRenderer
              key={item.key}
              item={item}
              itemHeight={itemHeight}
              itemWidth={itemWidth}
              handlePress={props.handlePress || (() => {})}
              divider={index !== props.items.length - 1}
            />
          );
        })}
      </View>
    </View>
  );
};

const ItemRenderer = ({
  item,
  itemHeight,
  itemWidth,
  divider,
  handlePress,
}: {
  item: ItemProps;
  itemHeight: number;
  itemWidth: number;
  divider: boolean;
  handlePress: (item: ProductProps) => void;
}) => {
  const theme = useTheme();
  const intl = useIntl();
  return (
    <Pressable onPress={() => handlePress(item)}>
      <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
        <View style={{flexDirection: 'row'}}>
          <Image
            testID="basketProductImage"
            source={{uri: item.image}}
            style={{height: itemHeight, width: itemWidth}}
          />

          <View
            style={{
              marginHorizontal: 15,
            }}>
            <Text
              testID="productName"
              variant="titleLarge"
              style={{
                marginBottom: 5,
                fontSize: 14,
                color: theme.colors.titleTextColor,
                fontWeight: '900',
              }}>
              {item.name}
            </Text>
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                marginBottom: 5,
              }}>
              <Text
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.titleTextColor,
                  fontWeight: '900',
                }}>
                {intl.formatMessage({defaultMessage: 'Quantity'}) + ' :'}
              </Text>

              <Text
                testID="productPrice"
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.greenTextColor,
                  fontWeight: '700',
                  marginLeft: 5,
                }}>
                {item.numberOfItems}
              </Text>
            </View>

            <Text
              variant="titleLarge"
              style={{
                fontSize: 13,
                color: theme.colors.titleTextColor,
                fontWeight: '600',
                marginBottom: 5,
              }}>
              {intl.formatNumber(item.price, {
                style: 'currency',
                currency: item.currency,
              }) +
                ' / ' +
                intl.formatMessage({defaultMessage: 'Piece'})}
            </Text>

            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                marginBottom: 5,
              }}>
              <Text
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.titleTextColor,
                  fontWeight: '900',
                }}>
                {intl.formatMessage({defaultMessage: 'Total'}) + ' :'}
              </Text>

              <Text
                testID="productPrice"
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.greenTextColor,
                  fontWeight: '700',
                  marginLeft: 5,
                }}>
                {intl.formatNumber(item.price * item.numberOfItems, {
                  style: 'currency',
                  currency: item.currency,
                })}
              </Text>
            </View>
          </View>
        </View>
      </View>
      {divider && <Divider style={{marginVertical: 10}} />}
    </Pressable>
  );
};

export default OrderedItems;

export const OrderedItemsCMSInput = {
  _component: OrderDetailsPageComponentType.OrderedItems,
  component: OrderedItems,
};
