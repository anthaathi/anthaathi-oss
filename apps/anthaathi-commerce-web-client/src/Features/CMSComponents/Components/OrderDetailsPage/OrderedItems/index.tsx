import { useStyletron } from '@anthaathi/solid-styletron';
import { For } from 'solid-js';

export interface ItemProps {
  id: number;
  name: string;
  description?: string;
  price: number;
  currency: string;
  image: string;
  weight_unit: string;
  packaging: string;
  key: string;
  numberOfItems: number;
  notes?: string;
}

export interface OrderedItemProps {
  title?: string;
  items: ItemProps[];
  handlePress?: (item: ItemProps) => void;
}

function OrderedItems(props: OrderedItemProps) {
  const [css, $theme] = useStyletron();
  return (
    <div
      class={css({
        margin: '0 auto',
        marginTop: $theme.sizing.scale800,
        marginBottom: $theme.sizing.scale800,
      })}
    >
      <For each={props.items}>{(item) => <ItemRenderer item={item} />}</For>
    </div>
  );
}

export default OrderedItems;

const ItemRenderer = ({ item }: { item: ItemProps }) => {
  const [css, $theme] = useStyletron();
  return (
    <div
      class={css({
        border: '1px solid #F0F0F0',
        borderRadius: '2px',
        backgroundColor: '#fff',
        paddingLeft: $theme.sizing.scale600,
        paddingRight: $theme.sizing.scale600,
        paddingTop: $theme.sizing.scale500,
        paddingBottom: $theme.sizing.scale500,
        marginBottom: $theme.sizing.scale600,
        display: 'flex',
      })}
    >
      <img
        src={item.image}
        class={css({
          height: '120px',
          width: '120px',
          objectFit: 'cover',
        })}
      />
      <div
        class={css({
          marginLeft: $theme.sizing.scale400,
        })}
      >
        <p
          class={css({
            ...$theme.typography.LabelLarge,
            fontWeight: 'bold',
            color: '#364A15',
            marginTop: 0,
            marginBottom: $theme.sizing.scale300,
          })}
        >
          {item.name}
        </p>
        <p
          class={css({
            ...$theme.typography.LabelMedium,
            fontWeight: 'normal',
            color: '#364A15',
            marginTop: 0,
            marginBottom: $theme.sizing.scale100,
          })}
        >
          {item.price + ' / Piece'}
        </p>
        <FlexText label1="Quantity" label2={item.numberOfItems} />
        <FlexText label1="Total" label2={item.numberOfItems * item.price} />
      </div>
    </div>
  );
};

const FlexText = ({
  label1,
  label2,
}: {
  label1: string;
  label2: string | number;
}) => {
  const [css, $theme] = useStyletron();
  return (
    <div
      class={css({
        display: 'flex',
      })}
    >
      <p
        class={css({
          ...$theme.typography.LabelLarge,
          fontWeight: 'bold',
          color: '#364A15',
          marginTop: 0,
          marginBottom: $theme.sizing.scale300,
        })}
      >
        {label1 + ': '}
      </p>

      <p
        class={css({
          ...$theme.typography.LabelLarge,
          fontWeight: 'bold',
          color: '#008D3E',
          marginTop: 0,
          marginBottom: $theme.sizing.scale300,
        })}
      >
        {label2}
      </p>
    </div>
  );
};
