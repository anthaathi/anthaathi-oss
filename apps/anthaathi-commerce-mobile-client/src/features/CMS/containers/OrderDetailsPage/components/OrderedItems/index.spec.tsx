import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import OrderedItems from '.';

describe('OrderedItems', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <OrderedItems
            title="Items"
            items={[
              {
                id: 1,
                name: 'Baby Yellow Pepper',
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                key: '12',
                price: 12,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
              },
              {
                id: 2,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('orderedItems')).toBeTruthy();
  });
});
