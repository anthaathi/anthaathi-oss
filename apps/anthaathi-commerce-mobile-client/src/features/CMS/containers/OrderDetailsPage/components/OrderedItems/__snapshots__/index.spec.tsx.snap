// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedItems should render the item 1`] = `
<View
  style={
    {
      "marginHorizontal": 10,
    }
  }
  testID="orderedItems"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "color": undefined,
          "fontSize": 16,
          "fontWeight": "600",
          "marginBottom": 5,
        }
      }
    >
      Items
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": undefined,
        "borderRadius": 4,
        "borderWidth": 1,
        "paddingHorizontal": 5,
        "paddingVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
              }
            }
            style={
              {
                "height": 250,
                "width": 240,
              }
            }
            testID="basketProductImage"
          />
          <View
            style={
              {
                "marginHorizontal": 15,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": undefined,
                    "fontSize": 14,
                    "fontWeight": "900",
                    "marginBottom": 5,
                  },
                ]
              }
              testID="productName"
              variant="titleLarge"
            >
              Baby Yellow Pepper
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Quantity :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                2
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": undefined,
                    "fontSize": 13,
                    "fontWeight": "600",
                    "marginBottom": 5,
                  },
                ]
              }
              variant="titleLarge"
            >
              $12.00 / Piece
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Total :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                $24.00
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            {
              "marginVertical": 10,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1",
              }
            }
            style={
              {
                "height": 250,
                "width": 240,
              }
            }
            testID="basketProductImage"
          />
          <View
            style={
              {
                "marginHorizontal": 15,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": undefined,
                    "fontSize": 14,
                    "fontWeight": "900",
                    "marginBottom": 5,
                  },
                ]
              }
              testID="productName"
              variant="titleLarge"
            >
              Capsicum mixed
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Quantity :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                2
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": undefined,
                    "fontSize": 13,
                    "fontWeight": "600",
                    "marginBottom": 5,
                  },
                ]
              }
              variant="titleLarge"
            >
              $23.00 / Piece
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Total :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                $46.00
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
